### PR TITLE
Test and fix using namespace from env file

### LINKED ISSUE #5544: Commands should get namespace from env file
/kind bug

## What versions of software are you using?

**Operating System:**

odo v2.5.0 (84a11447a)
**Output of `odo version`:**

## How did you run odo exactly?

The default namespace is prj1:

```
$ kubectl get kubectl get pods
No resources found in prj1 namespace.

$ kubectl get kubectl get pods -n prj2
No resources found in prj1 namespace.
```

ENV file defines the namespace as prj2:
```
$ cat .odo/env/env.yaml
ComponentSettings:
  Name: my-nodejs-app
  Project: prj2
  AppName: app
```

Running `odo deploy`

## Actual behavior

```
$ kubectl get pods -n prj1
NAME                       READY   STATUS    RESTARTS   AGE
my-node-78d595bddd-lfswt   1/1     Running   0          6s

$ kubectl get pods -n prj2
No resources found in prj2 namespace.
```

## Expected behavior

The component should be deployed in prj2, not prj1

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug
/kind tests

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #5544 

**PR acceptance criteria:**

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by feloy** [comment]
@kadel could you please check that these tests respect the specs?

File: tests/integration/devfile/cmd_devfile_deploy_test.go [hunk 1]
```diff
@@ -37,23 +38,55 @@ var _ = Describe("odo devfile deploy command tests", func() {
...
-		It("should run odo deploy", func() {
-			stdout := helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
-			By("building and pushing image to registry", func() {
-				Expect(stdout).To(ContainSubstring("build -t quay.io/unknown-account/myimage -f " + filepath.Join(commonVar.Context, "Dockerfile ") + commonVar.Context))
-				Expect(stdout).To(ContainSubstring("push quay.io/unknown-account/myimage"))
+
+		When("running odo deploy", func() {
+			var stdout string
+			BeforeEach(func() {
+				stdout = helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
```
Comment (valaparthvi) [inline tests/integration/devfile/cmd_devfile_deploy_test.go:56]: You haven't added the `env.yaml` exists, and the project check here.

File: tests/integration/devfile/cmd_devfile_deploy_test.go [hunk 2]
```diff
@@ -37,23 +38,55 @@ var _ = Describe("odo devfile deploy command tests", func() {
...
+
+			When("deleting previous deployment and switching kubeconfig to another namespace", func() {
+				var otherNS string
+				BeforeEach(func() {
+					helper.Cmd("odo", "delete", "component", "--name", cmpName, "-f").ShouldPass()
+					output := commonVar.CliRunner.Run("get", "deployment", "-n", commonVar.Project).Err.Contents()
+					Expect(string(output)).To(ContainSubstring("No resources found in " + commonVar.Project + " namespace."))
+
+					otherNS = commonVar.CliRunner.CreateRandNamespaceProject()
+					commonVar.CliRunner.SetProject(otherNS)
```
Comment (valaparthvi) [inline tests/integration/devfile/cmd_devfile_deploy_test.go:77]: `CreateRandNamespaceProject` creates and sets the project name.

File: tests/integration/devfile/cmd_dev_test.go [hunk 3]
```diff
@@ -165,5 +167,41 @@ var _ = Describe("odo dev command tests", func() {
...
+
+			When("deleting previous deployment and switching kubeconfig to another namespace", func() {
+				var otherNS string
+				BeforeEach(func() {
+					helper.Cmd("odo", "delete", "component", "--name", cmpName, "-f").ShouldPass()
+					output := commonVar.CliRunner.Run("get", "deployment", "-n", commonVar.Project).Err.Contents()
+					Expect(string(output)).To(ContainSubstring("No resources found in " + commonVar.Project + " namespace."))
+
+					otherNS = commonVar.CliRunner.CreateRandNamespaceProject()
+					commonVar.CliRunner.SetProject(otherNS)
```
Comment (valaparthvi) [inline tests/integration/devfile/cmd_dev_test.go:189]: Same comment as below.

**Review by valaparthvi — CHANGES_REQUESTED** [review]

**Review by valaparthvi — APPROVED** [review]

**Pull request merged**